Automattic/wp-calypso
# Domains: Adapt domain registrant verification page to be used with DSAPI reseller domains

## Proposed Changes

This PR updates the domain registrant verification landing page (`/domain-services/registrant-verification/`) to be used with DSAPI reseller domains.

A new `reseller=reseller-name` query parameter can be added to the URL. When that's added, references to WPCOM links and support pages are removed. If the reseller is `tumblr`, the Tumblr logo is shown in place of the WordPress logo.

### Screenshots

- Success

<img width="682" alt="Screenshot 2023-04-11 at 20 46 10" src="https://user-images.githubusercontent.com/5324818/231313855-3ce101d9-4b24-490c-9063-598a121c395b.png">

- Email already verified

<img width="684" alt="Screenshot 2023-04-11 at 21 02 58" src="https://user-images.githubusercontent.com/5324818/231313892-91f26fdd-a92d-4e33-b398-7c42bdcdddbe.png">

- Expired email

<img width="675" alt="Screenshot 2023-04-14 at 14 56 47" src="https://user-images.githubusercontent.com/5324818/232122236-aa1a36fb-8a84-4f62-9836-cedf524f0333.png">

- Mismatched email address

<img width="674" alt="Screenshot 2023-04-11 at 20 56 18" src="https://user-images.githubusercontent.com/5324818/231313945-0fe87fd1-d079-4855-bcf4-28bab84e676c.png">

- Error message

<img width="674" alt="Screenshot 2023-04-11 at 20 45 31" src="https://user-images.githubusercontent.com/5324818/231313960-b710371b-3268-4ee2-88bc-460a8ae818ae.png">

## Testing Instructions

- Open the live Calypso link or build this branch locally
- Hack your backend's `/domains/{domain}/verify-email` endpoint to return different success and error responses
- Access the registrant verification landing page with some example query parameters, e.g. `/domain-services/registrant-verification?domain=example.com&email=example@example.com&token=1234&reseller=tumblr_ote`
- Ensure the different backend responses' show pages as shown in the screenshots above

## Pre-merge Checklist
